server/asset/btc/btc.go

```diff
@@ -188,12 +188,49 @@ func ReadCloneParams(cloneParams interface{}) (*chaincfg.Params, error) {
 	return p, nil
 }
 
-// Coin is part of the asset.Backend interface. See the unexported Backend.utxo
-// method for the full implementation.
-func (btc *Backend) Coin(coinID []byte, redeemScript []byte) (asset.Coin, error) {
+// Contract is part of the asset.Backend interface. An asset.Contract is a
+// utxo that has been validated as a swap contract for the passed redeem script.
+func (btc *Backend) Contract(coinID []byte, redeemScript []byte) (asset.Contract, error) {
 	txHash, vout, err := decodeCoinID(coinID)
 	if err != nil {
-		return nil, fmt.Errorf("error decoding tx ID %x: %v", coinID, err)
+		return nil, fmt.Errorf("error decoding coin ID %x: %v", coinID, err)
+	}
+	utxo, err := btc.utxo(txHash, vout, redeemScript)
+	if err != nil {
+		return nil, err
+	}
+	err = utxo.auditContract()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Referring to my other comment, it seems like we need error types so we can recognize when `Contract` errors out because of asset network latency vs. contract issues.

Reply:
Maybe a `TxnNotFoundOnNetwork` (ok, bad name) sort of error for the try again case?

Reply:
Or a third output arg for `Contract` saying if utxo found / other failure.